### PR TITLE
Stream API cleanup

### PR DESCRIPTION
**Pull request by 0x213F**
**Summary**

Cleanup from previous string of PRs (https://github.com/0x213F/jukebox-radio-django/pull/9).

Inline comments for explanations of everything.

### CODE REVIEW

File: jukebox_radio/streams/migrations/0005_auto_20210117_2105.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Generated by Django 3.1.2 on 2021-01-17 21:05
+
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+class Migration(migrations.Migration):
```
Comment (0x213F) [inline jukebox_radio/streams/migrations/0005_auto_20210117_2105.py:7]: Just some quick model changes. It should have been this way in the beginning.

File: jukebox_radio/streams/models/queue_interval_model.py [hunk 2]
```diff
@@ -69,9 +77,13 @@ def highest_bound(self, queue_id):
 
     def conflicting_interval(self, *, queue_id, lower_bound_id, upper_bound_id):
         Marker = apps.get_model('streams', 'Marker')
+        Queue = apps.get_model('streams', 'Queue')
+
+        queue = Queue.objects.select_related("track").get(uuid=queue_id)
+        track = queue.track
 
-        lower_bound = Marker.objects.get(uuid=lower_bound_id, queue_id=queue_id)
```
Comment (0x213F) [inline jukebox_radio/streams/models/queue_interval_model.py:73]: This was wrong before - there is no queue field on Marker. We need to get the track from queue and use it to query for the bound markers.

File: jukebox_radio/streams/models/queue_interval_model.py [hunk 3]
```diff
@@ -84,6 +96,11 @@ def conflicting_interval(self, *, queue_id, lower_bound_id, upper_bound_id):
                     queue_id=queue_id,
                     upper_bound__timestamp_ms__gt=lower_bound.timestamp_ms,
                     upper_bound__timestamp_ms__lt=upper_bound.timestamp_ms,
+                ) |
+                Q(
```
Comment (0x213F) [inline jukebox_radio/streams/models/queue_interval_model.py:100]: Missing a case here: when a duplicate is created (i.e. has the same bounds).

File: jukebox_radio/streams/models/queue_interval_model.py [hunk 4]
```diff
@@ -110,7 +127,7 @@ class QueueInterval(models.Model):
 
     queue = models.ForeignKey(
         "streams.Queue",
-        related_name="+",
+        related_name="intervals",
```
Comment (0x213F) [inline jukebox_radio/streams/models/queue_interval_model.py:130]: This needs to be exposed

File: jukebox_radio/streams/models/queue_model.py [hunk 5]
```diff
@@ -13,18 +13,18 @@ class QueueManager(models.Manager):
...
         Track = apps.get_model("music", "Track")
+        QueueInterval = apps.get_model("streams", "QueueInterval")
 
         if not queue:
             return None
 
-        collection = (
-            Collection.objects.serialize(queue.collection) if not is_child else None
-        )
+        track = Track.objects.serialize(queue.track) if queue.track_id else None
```
Comment (0x213F) [inline jukebox_radio/streams/models/queue_model.py:21]: Small optimization

File: jukebox_radio/streams/models/queue_model.py [hunk 6]
```diff
@@ -40,6 +40,15 @@ def serialize(self, queue, is_child=False):
             children.append(self.serialize(child, is_child=True))
         obj["children"] = children
 
+        try:
```
Comment (0x213F) [inline jukebox_radio/streams/models/queue_model.py:43]: Need to expose interval data in queue data.

**Review by op1346 — APPROVED** [review]

**Pull request merged**